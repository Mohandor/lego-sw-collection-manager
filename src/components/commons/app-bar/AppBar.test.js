import React from 'react';
import { Grid, Typography, Toolbar, AppBar, Tabs, Tab } from '@material-ui/core';
import CustomAppBar from './AppBar';
import { shallow } from 'enzyme';

describe('<AppBar />', () => {
  it('should render the AppBar', () => {
    const props = {
      location: {
        pathname: '/minifigs'
      },
      classes: {
        fullHeight: 'fullHeight',
      }
    }
    const wrapper = shallow(<CustomAppBar {...props} />);
    expect(wrapper.find(Grid)).toHaveLength(1);
    expect(wrapper.find(Typography)).toHaveLength(1);
    expect(wrapper.find(Toolbar)).toHaveLength(1);
    expect(wrapper.find(AppBar)).toHaveLength(1);
    expect(wrapper.find(Tabs)).toHaveLength(1);
    expect(wrapper.find(Tab)).toHaveLength(2);
  });
});