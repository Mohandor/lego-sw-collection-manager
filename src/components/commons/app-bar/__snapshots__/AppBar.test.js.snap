// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBar /> should render the AppBar 1`] = `
<WithStyles(ForwardRef(AppBar))
  color="default"
  elevation={0}
  position="sticky"
>
  <WithStyles(ForwardRef(Toolbar))
    variant="dense"
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      justify="space-between"
    >
      <WithStyles(ForwardRef(Typography))
        className="flex"
        color="inherit"
        variant="h6"
      >
        <img
          alt="app-logo"
          className="appLogo"
          src="elephant.png"
        />
        <Hidden
          smDown={true}
        >
          Lego SW Collection Manager
        </Hidden>
      </WithStyles(ForwardRef(Typography))>
      <Hidden
        mdDown={true}
      >
        <Connect(withRouter(TabMenu)) />
      </Hidden>
      <Hidden
        lgUp={true}
      >
        <Connect(withRouter(BurgerNav)) />
      </Hidden>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;
